### PR TITLE
Fix MOAB canonical ordering during element splitting

### LINKED ISSUE #166: Negative volumes in source meshes
Source mesh elements have been observed to sometimes have negative volume, particularly near the magnetic axis where wedge splitting occurs. Investigation into the source_mesh.py code has revealed that the implementation of MOAB canonical ordering is incorrect for wedges.

### PR DESCRIPTION
**Pull request by connoramoreno**
Corrects implementation of MOAB canonical ordering. Changes are as follows:

- Changes MOAB canonical ordering for both wedges and hexahedra to coincide with element splitting shown in ParaStell publication (Moreno, Bader, Wilson, 2024)
- Adds documentation explaining MOAB canonical ordering for posterity

Closes #166

### CODE REVIEW

**Comment by connoramoreno** [comment]
I'm not sure how we'd test canonical ordering of tetrahedra directly but one test that we should probably add is to ensure no elements with negative volume are present

File: tests/test_source_mesh.py [hunk 1]
```diff
@@ -68,6 +70,31 @@ def test_vertices(source_mesh):
...
+    ) + tets_per_hex * (num_s - 2) * (num_theta - 1) * (num_phi - 1)
+    num_neg_vols_exp = 0
+
+    remove_files()
+
+    source_mesh.create_vertices()
+    source_mesh.create_mesh()
+
+    assert len(source_mesh.volumes) == num_elements_exp
+    assert len([i for i in source_mesh.volumes if i < 0]) == num_neg_vols_exp
```
Comment (gonuke) [inline tests/test_source_mesh.py:93]: Nice - does this fail with the old/wrong canonical ordering?

**Review by gonuke — COMMENTED** [review]
A quick question before we move forward....

File: tests/test_source_mesh.py (hunk 1 above)
Comment (connoramoreno) [reply tests/test_source_mesh.py:93]: With the old canonical ordering, this doesn't fail on the test equilibrium, but it does fail with WISTELL-D. On WISTELL-D using these parameters, the number of elements is as expected but 108 elements (1.47%) have negative volume.

**Review by gonuke — APPROVED** [review]
Thanks @connoramoreno

**Pull request merged**